arelle/ModelXbrl.py

```diff
@@ -10,7 +10,8 @@
 from decimal import Decimal
 from arelle import UrlUtil, XmlUtil, ModelValue, XbrlConst, XmlValidate
 from arelle.FileSource import FileNamedStringIO
-from arelle.ModelDtsObject import ModelRelationship
+from arelle.ModelRelationshipSet import ModelRelationshipSet
+from arelle.ModelRelationshipSet import ModelRelationshipSet as ModelRelationshipSetReturn
```

Comment:
If I use `ModelRelationshipSet` from `from arelle.ModelRelationshipSet import ModelRelationshipSet` as a return type:
![image](https://user-images.githubusercontent.com/21142447/183722211-df137421-1945-4159-8464-1a6385884105.png)

If I use `ModelRelationshipSetReturn` from `from arelle.ModelRelationshipSet import ModelRelationshipSet as ModelRelationshipSetReturn` as a return type:
![image](https://user-images.githubusercontent.com/21142447/183722339-6cb347c3-e6a1-4e4b-8060-0a8c8bef3f02.png)

I'm trying to achieve the latter, ie proper type hints for `fromModelObjects`.

Reply:
I'm assuming that's because `ModelRelationshipSet` is set to None further below?

Reply:
Yes, commenting out `ModelRelationshipSet = None # dynamic import` gives us a correct return type.

What about creating a `TYPE_MODEL_RELATIONSHIP_SET = ModelRelationshipSet` after all the imports and then using the constant as a return type? It yields the desired result.

Reply:
The variable name isn't what's important here. The dynamic import is (presumably) trying to avoid a circular dependency so loading the module at import time will cause problems regardless of what it's called.

That being said, there's no harm in importing a module multiple times (it'll simple return the existing imported module), so the existing `ModelRelationshipSet = None` pattern is unnecessary.

Use [typing.TYPE_CHECKING](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) in the header:
```python
if TYPE_CHECKING:
    from arelle import ModelRelationshipSet
```

and [remove the `ModelRelationshipSet = None` assignment](https://github.com/Arelle/Arelle/blob/451b9484d7453f71bb6b20935aa3365125bc1248/arelle/ModelXbrl.py#L23) along with unconditionally importing `ModelRelationshipSet` [from within `relationshipSet(...)`](https://github.com/Arelle/Arelle/blob/451b9484d7453f71bb6b20935aa3365125bc1248/arelle/ModelXbrl.py#L397-L399)

Reply:
Thanks! I had to do the import for `TYPE_CHECKING` like so: `from arelle.ModelRelationshipSet import ModelRelationshipSet`, otherwise type hints didn't work.